CONTRIBUTING.md

```diff
@@ -0,0 +1,31 @@
+# Contributing to Rust IPFS
+
+Welcome, and thank you for your interest in contributing to Rust IPFS. Issues and pull requests are encouraged.
+
+## Target Build
+
+Rust IPFS will always target the current _stable_ version of Rust that is released. Our CI/CD tests will reflect this.
+
+TODO: Instructions on installing the up-to-date stable toolchain
+TODO: First Principles
+
+## Roadmap
+
+- Link to roadmap issue or project?
+
+## Contributing
+
+Reporting Bugs
+Suggesting Enhancements
+Your First Code Contribution
+Pull Requests
+
+## TODO: Styleguides
+
+Git Commit Messages
+JavaScript Styleguide
+CoffeeScript Styleguide
+Specs Styleguide
+Documentation Styleguide
+
```

Comment:
There isn't uniform commit message style, like, I try to use the "[angularjs style](https://gist.github.com/stephenparish/9941e89d80e2bc58a153)" but make mistakes and the same time wouldn't want to start checking the messages like `js-ipfsd-ctl` does for example.

I think this section could be replaced with discussion on what the CI pipeline does: It runs `cargo fmt` and `cargo clippy`.

Reply:
From the AngularJS style message comes the https://www.conventionalcommits.org/. The nice thing about it is that you can generate changelogs from them. That's really nice especially for breaking changes.

Reply:
I didn't know a spec had been formed, probably worth linking to it.